### DISCUSSION TITLE
Well... It has finally happened, Anyone implement MQTT and HTTP REST API using asyncio as foundation?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16505] [General]
Well... It has finally happened, Anyone implement MQTT and HTTP REST API using asyncio as foundation?  I hope to make the code base transportable across ESP32 and ESP8266, but been a while since I worked with asyncio, last time I had a MQTT based implementation but now need HTTP REST API as well.  Yes, have to different types of clients.  Hoping someone has tackled this multiple client type scenario of I don't have to invent a wheel here.

**Comment by jonnor** [comment]
https://github.com/peterhinch/micropython-mqtt
https://github.com/micropython/micropython-lib/tree/master/micropython/uaiohttpclient
